### PR TITLE
New patch release SLEPc 3.17.1

### PR DESCRIPTION
**Pull request by joseeroman**
Contains also a new patch version of 3.16

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-slepc4py/package.py [hunk 1]
```diff
@@ -11,13 +11,15 @@ class PySlepc4py(PythonPackage):
...
 
     homepage = "https://gitlab.com/slepc/slepc4py"
-    url      = "https://slepc.upv.es/download/distrib/slepc4py-3.16.2.tar.gz"
+    url      = "https://slepc.upv.es/download/distrib/slepc4py-3.17.1.tar.gz"
     git      = "https://gitlab.com/slepc/slepc.git"
 
     maintainers = ['joseeroman', 'balay']
 
     version('main', branch='main')
+    version('3.17.1', sha256='967d5d045526088ff5b7b2cde76f8b4d1fee3a2a68481f85224b0795e6613eb9 ')
```
Comment (balay) [inline var/spack/repos/builtin/packages/py-slepc4py/package.py]:
```suggestion
    version('3.17.1', sha256='967d5d045526088ff5b7b2cde76f8b4d1fee3a2a68481f85224b0795e6613eb9')
```

File: var/spack/repos/builtin/packages/py-slepc4py/package.py [hunk 2]
```diff
@@ -11,13 +11,15 @@ class PySlepc4py(PythonPackage):
     """
 
     homepage = "https://gitlab.com/slepc/slepc4py"
-    url      = "https://slepc.upv.es/download/distrib/slepc4py-3.16.2.tar.gz"
+    url      = "https://slepc.upv.es/download/distrib/slepc4py-3.17.1.tar.gz"
```
Comment (balay) [inline var/spack/repos/builtin/packages/py-slepc4py/package.py:14]: BTW: this update is not needed [as the URL is just the template . It just adds to code churn - for each such update]

File: var/spack/repos/builtin/packages/py-slepc4py/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-slepc4py/package.py]: This fix is required (to resolve unit test failure).

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed all new sha256. 

Agree with @balay on needed changes, one of which is required by failing unit tests.

File: var/spack/repos/builtin/packages/py-slepc4py/package.py (hunk 2 above)
Comment (joseeroman) [reply var/spack/repos/builtin/packages/py-slepc4py/package.py:14]: Ok

**Review by balay — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**